### PR TITLE
Refactor module manager

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Related PRs       | [30926](https://github.com/PrestaShop/PrestaShop/pull/30926)
| How to test?      | Check that module manager works the same, but there are no empty categories in the category dropdown.
| Possible impacts? | 
| Sponsor company   | [TRENDO s.r.o.](https://www.trendo.cz)

### Changes
- Removed old code using `categoryParentEnglishName`. This not used anywhere in the core, in any native or custom module I saw. Related tests removed.
- Added comments to `CategoryProvider` class and improved clarity of a function getting the category.
- Added `{% if subMenu.modules is not empty %} `to module category dropdown, so there are no empty categories in the dropdown. They were removed from the list by @eternoendless previously in https://github.com/PrestaShop/PrestaShop/pull/29421.
- On PHP side, it was removing `CATEGORY_THEME` (Theme modules) if it was empty. I removed this code so branded distros can also use the hook to present their modules in the empty category. It will still be empty visually, there will be no change. I added this category to expected answer in tests.

Snímek obrazovky 2023-01-12 190526

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php [hunk 1]
```diff
@@ -64,8 +64,12 @@ class CategoriesProvider
 
     public function __construct(array $addonsCategories, array $modulesTheme)
     {
+        // List of modules (getModulesToEnable) present in the current theme's YML file
+        // We will use that to determine which modules will go to "Theme modules" category
         $this->modulesTheme = $modulesTheme;
-        // We now avoid calling the API. This data is loaded from a local YML file
+
+        // A list of categories and subcategories we got from local YML file
+        // This was previously fetched from addons
```
Comment (eternoendless) [inline src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php]:
```suggestion
        // In the past, this was fetched from addons marketplace
```

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php [hunk 2]
```diff
@@ -192,17 +198,16 @@ private function createMenuObject(string $menu, string $name, array $moduleIds =
...
     {
-        $moduleCategoryParent = $installedProduct->attributes->get('categoryParentEnglishName');
-        if (!isset($categories['categories']->subMenu[$moduleCategoryParent])) {
-            if (in_array($installedProduct->attributes->get('name'), $this->modulesTheme)) {
-                $moduleCategoryParent = self::CATEGORY_THEME;
-            } else {
-                $moduleCategoryParent = $this->getParentCategoryFromTabAttribute($installedProduct);
-            }
+        // If the module is on a list of "modules to enable" in current theme's YML file, they get
+        // a hardcoded special category
```
Comment (eternoendless) [inline src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php]:
```suggestion
        // a hardcoded special category "theme modules"
```

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php [hunk 3]
```diff
@@ -192,17 +198,16 @@ private function createMenuObject(string $menu, string $name, array $moduleIds =
...
+        if (in_array($installedProduct->attributes->get('name'), $this->modulesTheme)) {
+            return self::CATEGORY_THEME;
         }
 
-        if (array_key_exists($moduleCategoryParent, $categories['categories']->subMenu)) {
-            return $moduleCategoryParent;
+        // Let's get tab from the module file
```
Comment (eternoendless) [inline src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php]:
```suggestion
        // Look it up by tab attribute, as declared in the categories file
```

**Review by eternoendless — COMMENTED** [review]
Seems okay, just some suggestions to clarify comments

**Comment by Hlavtox** [comment]
@eternoendless Feedback applied 👍

**Review by eternoendless — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @Hlavtox 

Thank you for your PR, I tested it and it seems to works, as we can see here : 

image

image

Automated test failed as you said it, so we need @nesrineabdmouleh to update the test.

**Comment by nesrineabdmouleh** [comment]
Hello @Hlavtox @AureRita 

Thank you for the PR and the tests !
I create this [PR](https://github.com/PrestaShop/PrestaShop/pull/30926) to fix the failed automated tests :100:

The 2 PRs should be merged together :)
QA :heavy_check_mark: 

Thank you!

**Comment by kpodemski** [comment]
Thank you @Hlavtox @nesrineabdmouleh

**Pull request merged**